examples/src/main/java/com/mapbox/navigation/examples/core/ReRouteActivity.kt

```diff
@@ -0,0 +1,300 @@
+            startNavigation.visibility = View.GONE
+        }
+    }
+
+    override fun onStart() {
+        super.onStart()
+        mapView.onStart()
+        mapboxNavigation?.registerLocationObserver(locationObserver)
+    }
+
+    public override fun onResume() {
+        super.onResume()
+        mapView.onResume()
+    }
+
+    public override fun onPause() {
+        super.onPause()
+        mapView.onPause()
+    }
+
+    override fun onStop() {
+        super.onStop()
+        mapboxNavigation?.unregisterLocationObserver(locationObserver)
+        mapboxNavigation?.unregisterRouteProgressObserver(routeProgressObserver)
+        mapboxNavigation?.unregisterRoutesObserver(routeObserver)
+        stopLocationUpdates()
+        mapView.onStop()
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        mapboxNavigation?.stopTripSession()
+        mapboxNavigation?.onDestroy()
+        mapView.onDestroy()
+    }
+
+    override fun onLowMemory() {
+        super.onLowMemory()
+        mapView.onLowMemory()
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+        navigationMapboxMap?.saveStateWith(MAP_INSTANCE_STATE_KEY, outState)
+        mapView.onSaveInstanceState(outState)
+    }
+
+    override fun onRestoreInstanceState(savedInstanceState: Bundle?) {
+        super.onRestoreInstanceState(savedInstanceState)
+        mapInstanceState = savedInstanceState?.getParcelable(MAP_INSTANCE_STATE_KEY)
+    }
+
+    override fun onPermissionResult(granted: Boolean) {
+        when (granted) {
+            true -> {
+                // todo
+            }
+            false -> {
+                // todo
+            }
+        }
+    }
+
+    override fun onExplanationNeeded(permissionsToExplain: MutableList<String>?) {
+        Toast.makeText(this, "todo needs doing", Toast.LENGTH_LONG).show()
+    }
+
+    private val locationListenerCallback: LocationEngineCallback<LocationEngineResult> =
+            object : LocationEngineCallback<LocationEngineResult> {
+                override fun onSuccess(result: LocationEngineResult) {
+                    // todo
+                }
+
+                override fun onFailure(exception: Exception) {
+                    Timber.i(exception)
+                }
+            }
+
+    private fun stopLocationUpdates() {
+        locationEngine?.removeLocationUpdates(locationListenerCallback)
+    }
+
+    private val locationObserver = object : LocationObserver {
+        override fun onRawLocationChanged(rawLocation: Location) {
+            Timber.d("raw location %s", rawLocation.toString())
+        }
+
+        override fun onEnhancedLocationChanged(
+            enhancedLocation: Location,
+            keyPoints: List<Location>
+        ) {
+            mapView.getMapAsync { mapboxMap ->
+                if (keyPoints.isNotEmpty()) {
+                    mapboxMap.locationComponent.forceLocationUpdate(keyPoints, true)
```

Comment:
I don't know why it's confusing. To me it makes more sense to get the locationComponent from its source rather than storing a nullable and mutable reference to it. However if you prefer your suggestion for the sake of consistency in the examples I can change it.